libs/go/libp2p_node/dht/dhtpeer/dhtpeer_test.go

```diff
@@ -1959,7 +1959,6 @@ func ensureAddressAnnounced(peers ...*DHTPeer) {
 		for !peer.addressAnnounced {
 			select {
 			case <-ctx.Done():
-				break
```

Comment:
it's needed to exit connection loop
but you can set `err` here. so it will exit by err